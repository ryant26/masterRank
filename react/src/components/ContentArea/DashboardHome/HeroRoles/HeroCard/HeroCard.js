import React, {
  Component
} from 'react';

import HeroStatsList from '../HeroStatsList/HeroStatsList';
import HeroImage from '../../../../HeroImage/HeroImage';
import PropTypes from 'prop-types';
import Modal from '../../../../Modal/Modal';
import Model from '../../../../../model/model';

const classNames = require('classnames');


export default class HeroCard extends Component {

    constructor(props) {
        super(props);
        this.state = {
            isStatsToggleOn: false,
            modalOpen: false,
            invitable: props.hero.platformDisplayName !== props.user.platformDisplayName
        };

        this.toggleStats = this.toggleStats.bind(this);
        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
        this.invitePlayer = this.invitePlayer.bind(this);
    }

    toggleStats() {
        this.setState(prevState => ({
            isStatsToggleOn: !prevState.isStatsToggleOn
        }));
    }

    invitePlayer() {
        const userObject = {
            "platformDisplayName": this.props.hero.platformDisplayName,
            "heroName": this.props.hero.heroName
        };
        Model.inviteUserToGroup(userObject);
    }

    openModal() {
        this.setState(() => {
            return {
                modalOpen: true
            };
        });
    }

    closeModal() {
        this.setState(() => {
            return {
                modalOpen: false
            };
        });
    }

    render() {
        const classses = classNames({
            overlay: true,
            invitable: this.state.invitable,
        });

        let statLine;

        if(this.props.hero.stats) {
            let wins = this.props.hero.stats.wins;
            let losses = this.props.hero.stats.losses;
            let winPercentage = parseFloat(wins/(wins+losses) * 100.0).toFixed(1);
            statLine = (
                <div className="sub-title">
                    <span>{this.props.hero.skillRating} SR</span>
                    <span> | </span><span>{winPercentage}% WR</span>
                </div>
            );
        }


    return (
        <div className="HeroCard flex align-center">
            <div className="invitePlayerButton">
                <HeroImage heroName={this.props.hero.heroName}/>
                <div className={classses}>
                    <div className="plus-container flex align-center justify-center" onClick={this.invitePlayer}>
                        <i className="fa fa-plus"/>
                    </div>
                </div>
            </div>
            <div className="content flex flex-column">
                <div className="display-name">{this.props.hero.platformDisplayName}</div>
                {statLine}
            </div>
            <div className="button-primary" onClick={this.openModal}>
                <div className="button-content">
                    stats
                </div>
            </div>
            <Modal modalOpen={this.state.modalOpen} closeModal={this.closeModal}>
                <HeroStatsList hero={this.props.hero} invitable={this.state.invitable}/>
            </Modal>

        </div>
    );
    }
}

HeroCard.propTypes = {
    hero: PropTypes.shape({
        heroName: PropTypes.string.isRequired,
        platformDisplayName: PropTypes.string.isRequired,
        skillRating: PropTypes.number.isRequired,
        stats: PropTypes.shape({
            wins: PropTypes.number.isRequired,
            losses: PropTypes.number.isRequired,
        })
    }),
    user: PropTypes.shape({
        platformDisplayName: PropTypes.string.isRequired
    })
};