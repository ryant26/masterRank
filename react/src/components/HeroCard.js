import React, {
  Component
} from 'react';

import HeroCardStats from './HeroCardStats.js'
import HeroImage from './HeroImage.js'
import InvitePlayerButton from './InvitePlayerButton.js'

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleInvitePlayerButton = this.toggleInvitePlayerButton.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }))
  }

  toggleInvitePlayerButton() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }))
  }

  render() {
    const containerStyle = {
      'display': 'flex',
      'backgroundColor': '#66ccff',
      'margin': '10px',
    }

    const statsStyle = {
      'display': 'flex',
    }

    const componentStyle = {
      'textAlign': 'center',
      'height': '60px',
      'width': '33%',
    }

    let winPercentage = parseFloat(this.props.hero.general_stats.win_percentage * 100.0).toFixed(1);

    return (
      <div>
        <div style={containerStyle}
          onMouseEnter={this.toggleInvitePlayerButton}
          onMouseLeave={this.toggleInvitePlayerButton}>

          {
            this.state.isHovering
            ? <InvitePlayerButton />
            : <HeroImage heroName={this.props.hero.name}/>
          }

          <div> Username </div>

          <div style={statsStyle} onClick={this.toggleStats}>
            <div style={componentStyle} >
              <div>Rank</div>
              <div>2734</div>
            </div>

            <div style={componentStyle}>
              <div>Games Played</div>
              <div>{this.props.hero.general_stats.games_played}</div>
            </div>

            <div style={componentStyle}>
              <div>WIN %</div>
              <div>{winPercentage}%</div>
            </div>
          </div>
        </div>
        
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}
