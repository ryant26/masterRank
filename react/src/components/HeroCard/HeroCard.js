import React, {
  Component
} from 'react';

import HeroCardStats from '../HeroCardStats/HeroCardStats';
import HeroImage from '../HeroImage/HeroImage';
import InvitePlayerButton from '../InvitePlayerButton/InvitePlayerButton';
import PropTypes from 'prop-types';

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleInvitePlayerButton = this.toggleInvitePlayerButton.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }));
  }

  toggleInvitePlayerButton() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }));
  }

  render() {
    const backgroundStyle = {
      'backgroundColor': '#66ccff'      
    };

    const containerStyle = {
      'display': 'flex',
    };

    const statsStyle = {
      'display': 'flex',
      'flexDirection':'column',
      'width':'240px',
      'cursor':'s-resize'      
    };

    const componentStyle = {
      'height':'20px'
    };
    
    let wins = this.props.hero.wins;
    let losses = this.props.hero.losses;
    let winPercentage = parseFloat(wins/(wins+losses) * 100.0).toFixed(1);

    return (
      <div>
        <div style={backgroundStyle}>
          <div onMouseEnter={this.toggleInvitePlayerButton}
               onMouseLeave={this.toggleInvitePlayerButton}>
                {
                    this.state.isHovering? <InvitePlayerButton /> : 
                                           <HeroImage heroName={this.props.hero.heroName}/>
                }
          </div>
          <div style={containerStyle}>
            <div style={statsStyle} onClick={this.toggleStats}>
              <div>
                <div><b>{this.props.hero.platformDisplayName}</b></div>
              </div>
              <div style={componentStyle} >
                <div><b>Rank:</b> 2733</div>
              </div>

              <div style={componentStyle}>
                <div><b>Hours Played:</b> {this.props.hero.hoursPlayed}</div>
              </div>

              <div style={componentStyle}>
                <div><b>Win Rate:</b> {winPercentage}%</div>
              </div>
            </div>
          </div>
        </div>
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}

HeroCard.propTypes = {
  hero: PropTypes.shape({
    heroName: PropTypes.string.isRequired,
    wins: PropTypes.number.isRequired,
    losses: PropTypes.number.isRequired,
    platformDisplayName: PropTypes.string.isRequired,
    hoursPlayed: PropTypes.number.isRequired,
  })
};