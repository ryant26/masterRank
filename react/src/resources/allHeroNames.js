export const names = [
  'ana',
  'bastion',
  'dva',
  'genji',
  'hanzo',
  'junkrat',
  'lucio',
  'mccree',
  'mei',
  'mercy',
  'orisa',
  'pharah',
  'reaper',
  'reinhardt',
  'roadhog',
  'soldier76',
  'sombra',
  'symmetra',
  'torbjorn',
  'tracer',
  'widowmaker',
  'winston',
  'zarya',
  'zenyatta'
];
