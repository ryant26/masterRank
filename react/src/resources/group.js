const GROUP = [
    {
      "_id": 12,
      "platformDisplayName": "PwNShoPP#1662",
      "platform": "pc",
      "region": "us",
      "lastModified": "2017-10-28T06:08:42.474Z",
      "heroName": "soldier76",
      "hoursPlayed": 4,
      "wins": 10,
      "losses": 1,
      "kdRatio": 3.26,
      "accuracy": 43,
      "blockedPerMin": 0,
      "healingPerMin": 0,
      "damagePerMin": 1305.59,
      "avgObjElims": 9.16,
      "avgObjTime": 53.32,
      "pKdRatio": 0.75,
      "pAccuracy": 0.60,
      "pBlockedPerMin": 0,
      "pHealingPerMin": 0.25,
      "pDamagePerMin": 0.9,
      "pAvgObjElims": 0.5,
      "pAvgObjTime": 0.65,
    }, 
    {
      "_id": 12,
      "platformDisplayName": "santa#1662",
      "platform": "pc",
      "region": "us",
      "lastModified": "2017-10-28T06:08:42.474Z",
      "heroName": "dva",
      "wins": 10,
      "losses": 1,
      "kdRatio": 3.26,
      "hoursPlayed": 4,
      "accuracy": 43,
      "blockedPerMin": 0,
      "healingPerMin": 0,
      "damagePerMin": 1305.59,
      "avgObjElims": 9.16,
      "avgObjTime": 53.32,
      "pKdRatio": 0.75,
      "pAccuracy": 0.60,
      "pBlockedPerMin": 0,
      "pHealingPerMin": 0.25,
      "pDamagePerMin": 0.9,
      "pAvgObjElims": 0.5,
      "pAvgObjTime": 0.65,  },
    {
      "_id": 12,
      "platformDisplayName": "rustyK#1662",
      "platform": "pc",
      "region": "us",
      "lastModified": "2017-10-28T06:08:42.474Z",
      "heroName": "hanzo",
      "wins": 10,
      "losses": 1,
      "kdRatio": 3.26,
      "hoursPlayed": 4,
      "accuracy": 43,
      "blockedPerMin": 0,
      "healingPerMin": 0,
      "damagePerMin": 1305.59,
      "avgObjElims": 9.16,
      "avgObjTime": 53.32,
      "pKdRatio": 0.75,
      "pAccuracy": 0.60,
      "pBlockedPerMin": 0,
      "pHealingPerMin": 0.25,
      "pDamagePerMin": 0.9,
      "pAvgObjElims": 0.5,
      "pAvgObjTime": 0.65,  },
    {
      "_id": 12,
      "platformDisplayName": "wisesm0#1147",
      "platform": "pc",
      "region": "us",
      "lastModified": "2017-10-28T06:08:42.474Z",
      "heroName": "zenyatta",
      "wins": 10,
      "losses": 1,
      "kdRatio": 3.26,
      "hoursPlayed": 4,
      "accuracy": 43,
      "blockedPerMin": 0,
      "healingPerMin": 0,
      "damagePerMin": 1305.59,
      "avgObjElims": 9.16,
      "avgObjTime": 53.32,
      "pKdRatio": 0.75,
      "pAccuracy": 0.60,
      "pBlockedPerMin": 0,
      "pHealingPerMin": 0.25,
      "pDamagePerMin": 0.9,
      "pAvgObjElims": 0.5,
      "pAvgObjTime": 0.65,  },
  ];
  
  export default GROUP;
  