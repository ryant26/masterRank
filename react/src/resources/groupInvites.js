const groupInvites = [
  {
    groupId: 10,
    groupSize: 4,
    inviteDate: "2018-02-19T12:57:24.790Z",
    leader: {
      platformDisplayName: "luckbomb#1234",
      skillRating: 3200,
      heroName: "genji",
      stats: {}
    },
    members: [
      {
        platformDisplayName: "wismo0#1234",
        skillRating: 2200,
        heroName: "widowmaker",
        stats: {}
      }
    ],
    pending: [
      {
        platformDisplayName: "andyLin#1234",
        skillRating: 1800,
        heroName: "tracer",
        stats: {}
      }
    ]
  },
  {
    groupId: 8,
    groupSize: 3,
    inviteDate: "2017-09-19T21:57:24.790Z",
    leader: {
      platformDisplayName: "luckbomb#1234",
      skillRating: 3200,
      heroName: "genji",
      stats: {}
    },
    members: [],
    pending: [
      {
        platformDisplayName: "cutie#1320",
        skillRating: 1600,
        heroName: "tracer",
        stats: {}
      }
    ]
  },
    {
        groupId: 2,
        groupSize: 4,
        inviteDate: "2017-06-19T16:57:24.790Z",
        leader: {
            platformDisplayName: "luckbomb#1234",
            skillRating: 3200,
            heroName: "doomfist",
            stats: {}
        },
        members: [
            {
                platformDisplayName: "wismo0#1234",
                skillRating: 2200,
                heroName: "soldier76",
                stats: {}
            }
        ],
        pending: [
            {
                platformDisplayName: "cutie#1320",
                skillrating: 1600,
                heroName: "winston",
                stats: {}
            }
        ]
    },
     {
      groupId: 210,
      groupSize: 6,
      inviteDate: "2018-02-19T16:57:24.790Z",
      leader: {
        platformDisplayName: "luckbomb#1234",
        skillRating: 3200,
        heroName: "genji",
        stats: {}
      },
      members: [
        {
          platformDisplayName: "wismo0#1234",
          skillRating: 2200,
          heroName: "widowmaker",
          stats: {}
        },{
          platformDisplayName: "andyLin#1234",
          skillRating: 1600,
          heroName: "winston",
          stats: {}
        }
      ],
      pending: [
        {
          platformDisplayName: "cutie#1320",
          skillRating: 1800,
          heroName: "tracer",
          stats: {}
        },{
          platformDisplayName: "pwnShopp#4810",
          skillRating: 1800,
          heroName: "soldier76",
          stats: {}
        },{
          platformDisplayName: "banannaTruck#1872",
          skillRating: 1800,
          heroName: "doomfist",
          stats: {}
        }
      ]
    },
];

export default groupInvites;