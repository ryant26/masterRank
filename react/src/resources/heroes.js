const HEROES = [
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "soldier76",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "hanzo",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "dva",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "zenyatta",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
];

export default HEROES;
