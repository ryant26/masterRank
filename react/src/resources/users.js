export const users = [
    {
        "_id": 12,
        "platformDisplayName": "PwNShoPP#1662",
        "lastUpdated": "2017-10-15T01:43:36.459Z",
        "platform": "pc",
        "level": 155,
        "portrait": "https://d1u1mce87gyfbn.cloudfront.net/game/unlocks/0x025000000000115A.png",
        "region": "us"
    }
];