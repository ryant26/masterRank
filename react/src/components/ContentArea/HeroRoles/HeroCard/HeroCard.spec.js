import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import HEROS from '../../../../resources/heroes';
import {users} from '../../../../resources/users';
import HeroCard from './HeroCard';

describe('HeroCard Component',()=> {
    let getHeroConfig = function() {
        return Object.assign({}, HEROS[0]);
    };

    let getUser = function() {
        return Object.assign({}, users[0]);
    };

    it('should render without exploding', () => {
        const wrapper = mount(
            <HeroCard hero={getHeroConfig()} user={getUser()}/>
        );

        const HeroCardComponent = wrapper.find(HeroCard);
        expect(HeroCardComponent).toBeTruthy();
    });

    it('should not have the invitable class for the users own heroes', () => {
        const platformDisplayName = 'pwnshopp';
        let heroConfig = getHeroConfig();
        let user = getUser();

        heroConfig.platformDisplayName = platformDisplayName;
        user.platformDisplayName = platformDisplayName;

        const wrapper = mount (
            <HeroCard hero={heroConfig} user={user}/>
        );

        expect(wrapper.find('.invitable').length).toBe(0);
    });

    it('should have the invitable class for other users heroes', () => {
        const platformDisplayName = 'pwnshopp';
        let heroConfig = getHeroConfig();
        let user = getUser();

        heroConfig.platformDisplayName = platformDisplayName;
        user.platformDisplayName = 'not_pwnshopp';

        const wrapper = mount (
            <HeroCard hero={heroConfig} user={user}/>
        );

        expect(wrapper.find('.invitable').length).toBe(1);
    });

    it('should toggle the stats card on click', () => {
        const component = renderer.create(
            <HeroCard hero={getHeroConfig()} user={getUser()}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component.getInstance().toggleStats();  
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component.getInstance().toggleStats();  
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });
});