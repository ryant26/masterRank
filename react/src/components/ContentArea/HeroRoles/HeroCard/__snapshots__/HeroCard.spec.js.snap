// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard Component should toggle the stats card on click 1`] = `
<div
  className="HeroCard flex align-center"
>
  <div
    className="invitePlayerButton"
  >
    <img
      alt="soldier76 icon"
      className="HeroImage"
      onClick={undefined}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <div
      className="overlay"
    >
      <div
        className="plus-container flex align-center justify-center"
      >
        <i
          className="fa fa-plus"
        />
      </div>
    </div>
  </div>
  <div
    className="content flex flex-column"
  >
    <div
      className="display-name"
    >
      PwNShoPP#1662
    </div>
    <div
      className="sub-title"
    >
      <span>
        2400
         SR
      </span>
      <span>
         | 
      </span>
      <span>
        61.1
        % WR
      </span>
    </div>
  </div>
  <div
    className="button-primary"
    onClick={[Function]}
  >
    <div
      className="button-content"
    >
      stats
    </div>
  </div>
</div>
`;

exports[`HeroCard Component should toggle the stats card on click 2`] = `
<div
  className="HeroCard flex align-center"
>
  <div
    className="invitePlayerButton"
  >
    <img
      alt="soldier76 icon"
      className="HeroImage"
      onClick={undefined}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <div
      className="overlay"
    >
      <div
        className="plus-container flex align-center justify-center"
      >
        <i
          className="fa fa-plus"
        />
      </div>
    </div>
  </div>
  <div
    className="content flex flex-column"
  >
    <div
      className="display-name"
    >
      PwNShoPP#1662
    </div>
    <div
      className="sub-title"
    >
      <span>
        2400
         SR
      </span>
      <span>
         | 
      </span>
      <span>
        61.1
        % WR
      </span>
    </div>
  </div>
  <div
    className="button-primary"
    onClick={[Function]}
  >
    <div
      className="button-content"
    >
      stats
    </div>
  </div>
</div>
`;

exports[`HeroCard Component should toggle the stats card on click 3`] = `
<div
  className="HeroCard flex align-center"
>
  <div
    className="invitePlayerButton"
  >
    <img
      alt="soldier76 icon"
      className="HeroImage"
      onClick={undefined}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <div
      className="overlay"
    >
      <div
        className="plus-container flex align-center justify-center"
      >
        <i
          className="fa fa-plus"
        />
      </div>
    </div>
  </div>
  <div
    className="content flex flex-column"
  >
    <div
      className="display-name"
    >
      PwNShoPP#1662
    </div>
    <div
      className="sub-title"
    >
      <span>
        2400
         SR
      </span>
      <span>
         | 
      </span>
      <span>
        61.1
        % WR
      </span>
    </div>
  </div>
  <div
    className="button-primary"
    onClick={[Function]}
  >
    <div
      className="button-content"
    >
      stats
    </div>
  </div>
</div>
`;
